packages/react-ui/src/PhantomProvider.tsx

```diff
@@ -186,99 +174,36 @@ function PhantomUIProvider({ children, theme = "light", customTheme }: Omit<Phan
-                )}
-
-                {/* Primary auth options - Phantom, Google */}
-                {!isMobile && (
-                  <>
-                    {/* Login with Phantom (embedded provider using Phantom extension) */}
-                    {isPhantomLoginAvailable.isAvailable && (
-                      <button
-                        className="phantom-ui-provider-button phantom-ui-provider-button-primary"
-                        onClick={() => connectWithAuthProvider("phantom")}
-                        disabled={connectionState.isConnecting}
-                      >
-                        {connectionState.isConnecting && connectionState.providerType === "embedded"
-                          ? "Connecting..."
-                          : "Login with Phantom"}
-                      </button>
-                    )}
-
-                    {/* Continue with Google */}
-                    <button
-                      className="phantom-ui-provider-button"
-                      onClick={() => connectWithAuthProvider("google")}
-                      disabled={connectionState.isConnecting}
-                    >
-                      {connectionState.isConnecting && connectionState.providerType === "embedded"
-                        ? "Connecting..."
-                        : "Continue with Google"}
-                    </button>
-                  </>
-                )}
-
-                {/* Extension option - smaller UI section */}
-                {!isMobile && isExtensionInstalled.isInstalled && (
-                  <div className="phantom-ui-extension-section">
-                    <div className="phantom-ui-divider">
-                      <span>or</span>
-                    </div>
-                    <button
-                      className="phantom-ui-provider-button phantom-ui-provider-button-secondary"
-                      onClick={connectWithInjected}
-                      disabled={connectionState.isConnecting}
-                    >
-                      {connectionState.isConnecting && connectionState.providerType === "injected"
-                        ? "Connecting..."
-                        : "Continue with extension"}
-                    </button>
-                  </div>
-                )}
-              </div>
-            </div>
-          </div>
-        </div>
-      )}
+      <Modal
+        isVisible={connectionState.isVisible}
+        isConnecting={connectionState.isConnecting}
+        error={connectionState.error}
+        providerType={connectionState.providerType}
+        appIcon={appIcon}
+        appName={appName}
+        isMobile={isMobile}
+        isExtensionInstalled={isExtensionInstalled.isInstalled}
+        isPhantomLoginAvailable={isPhantomLoginAvailable.isAvailable}
+        onClose={hideConnectionModal}
+        onConnectWithDeeplink={connectWithDeeplink}
+        onConnectWithAuthProvider={connectWithAuthProvider}
+        onConnectWithInjected={connectWithInjected}
```

Comment:
Do we intend to aid testing by passing these directly? A lot of them are in the `PhantomUIContext`, which is also retrieved in the component.